torchdata/datapipes/iter/load/huggingface.py

```diff
@@ -17,42 +16,55 @@
 
 
 def _get_response_from_huggingface_hub(
-    dataset: str, split: str, revision: str, streaming: bool, data_files: Optional[Dict[str, str]]
+    dataset: str,
+    split: Union[str, datasets.Split] = "train",
+    revision: Union[str, datasets.Version] = "main",
```

Comment:
Let's keep the annotation as `str` to make sure `datasets` as optional dependency